### PR TITLE
Issue 31 - implement disconnect method

### PR DESCRIPTION
**Pull request by johndietz**
Please consider this pull request to address issue #31.

### CODE REVIEW

File: src/busterbunny.js [hunk 1]
```diff
@@ -305,6 +319,24 @@ module.exports = (function() {
...
+                    _connected = false;
+                }
+            }
+            catch(err)
+            {
+                _connected = false;
+            }
+            finally
+            {
+                self.emitStats();
```
Comment (enigmatic00) [inline src/busterbunny.js]: shouldn't this & the subsequent disconnected event only be emitted in the callback of close?

File: test/integration/test.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    });
+
+    it('should gracefully handle a disconnect if disconnect is called twice', function(done){
+        var calledCount = 0;
+        var busterBunny = new BusterBunny(config.amqp);
+        function onDisconnected(){
+            calledCount += 1;
+            console.log(calledCount);
+            busterBunny.disconnect();
+            done();
```
Comment (enigmatic00) [inline test/integration/test.js:33]: won't this test result in `done()` being called multiple times?

Comment (johndietz) [inline test/integration/test.js:33]: it doesn't. the "disconnected" event is only emitted when it truly disconnects, and will guard against duplicate calls.

File: src/busterbunny.js (hunk 1 above)
Comment (johndietz) [inline src/busterbunny.js]: very good point. i'll adjust this in the am.

File: package.json [hunk 3]
```diff
@@ -23,7 +23,7 @@
     "sinon": "^1.17.2"
   },
   "scripts": {
-    "test": "mocha",
+    "test": "mocha --recursive test/unit",
```
Comment (binaryalchemist) [inline package.json:26]: Please remove mocha.opts file from `/test` folder.

**Comment by binaryalchemist** [comment]
Fixes #31.

**Pull request merged**